pkg/dns/dns.go

```diff
@@ -282,6 +282,21 @@ func (h *Handler) ServeDNS(w dns.ResponseWriter, req *dns.Msg) {
 		h.handleDefault(w, req)
 	}
 }
+func StartWithListener(opts ServerOptions) (*Server, error) {
```

Comment:
It would be good if we add some documentation to this function:
- Note that the "Listener" in the function name refers to `opts.Listener`
- Say something about this always being treated as TCP (as opposed to UDP) for the DNS server.

Reply:
Nit: should have empty line before the function (at least the other functions in the file have it).